lib/metasploit/framework/password_crackers/jtr/formatter.rb

```diff
@@ -76,7 +79,9 @@ def hash_to_jtr(cred)
       # /ssha/
       # /raw-sha512/
       # /raw-sha256/
-      return "#{cred.public.username}:#{cred.private.data}:#{cred.id}:"
+      # this also handles *other* and it isn't guaranteed to have a public
```

Comment:
Need to be more specific here, this assumes too much background knowledge from the reader and the terms "other" and "it isn't" aren't clear here.

Reply:
I changed the wording here to `This also handles *other* type credentials which aren't guaranteed to have a public` but let me know if you think this needs changing.

Reply:
Addressed with https://github.com/rapid7/metasploit-framework/pull/16341/commits/1408ffbab7d8fa24fa0a6fd4ff6abcb7ca5dce04